CoreFoundation/String.subproj/CFString.c

```diff
@@ -1264,12 +1264,16 @@ CF_PRIVATE CFStringRef __CFStringCreateImmutableFunnel3(
 	contentsDeallocator = __CFGetDefaultAllocator();
     }
 
+    #ifndef __ANDROID__
+    // https://bugs.swift.org/browse/SR-2587
+    // crashes String.swift, line 51 when creating String from zero length Data
```

Comment:
I think I’ve been able to resolve this problem. Constant strings have a fake isa which wasn’t being picked up or linked correctly so I created an alias in the source. The Locale problem was just a manifestation of the same problem as it used CONST_STRING_DECL().

#ifdef __ANDROID__
// Avoids crashes on Android
// https://bugs.swift.org/browse/SR-2587
// https://bugs.swift.org/browse/SR-2588
// Seemed to be a linker/relocation? problem.
// CFStrings using CONST_STRING_DECL() were not working
// Applies reference to _NSCFConstantString's isa here
// rather than using a linker option to create an alias.
#define __CFConstantStringClassReference _TMC10Foundation19_NSCFConstantString
#endif